test/e2e/egress_test.go

```diff
@@ -38,7 +38,15 @@ import (
 	controllerconfig "antrea.io/antrea/pkg/config/controller"
 )
 
-const waitEgressRealizedTimeout = 3 * time.Second
+const (
+	egressRealizedTimeout = 5 * time.Second
+	// When controller and agent restarted, agent may fail to start watch for EgressGroup,
+	// because antrea service is still not reachable for the agent.
+	// Thus agent wait another 5 second and try to start watch for EgressGroup again.
+	// pkg/agent/controller/egress/egress_controller.go-wait.NonSlidingUntil(c.watchEgressGroup, 5*time.Second, stopCh)
+	// See https://github.com/antrea-io/antrea/issues/2945
+	egressIPRealizedTimeout = 10 * time.Second
```

Comment:
just one minor question, you tested this timeout in your setup, it always work, right?

Reply:
Yes, has been tested in the same testbed that the problem occurred.